plugins/node/instrumentation-fs/src/types.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import type * as fs from 'fs';
+
+import type * as api from '@opentelemetry/api';
+import type { InstrumentationConfig } from '@opentelemetry/instrumentation';
+
+export type FunctionPropertyNames<T> = {
+  [K in keyof T]: T[K] extends Function ? K : never;
+}[keyof T];
+export type FunctionProperties<T> = Pick<T, FunctionPropertyNames<T>>;
+
+export type FMember = FunctionPropertyNames<typeof fs>;
+export type FPMember = FunctionPropertyNames<typeof fs['promises']>;
+
+export interface FsInstrumentationConfig extends InstrumentationConfig {
+  createHook?: (
+    functionName: string,
+    info: { args: ArrayLike<unknown> }
+  ) => boolean;
+  endHook?: (
```

Comment:
I think in other places we usually set the first param to `span` and place function data in the second object

Reply:
I've prioritized consistency between the hooks over consistency between the other instrumentations here. But it does make sense to have Span up front for `endHook` - We have it the other way around as well(`responseHook` in aws sdk). Span is not available for `createHook`.

This would be an option:

```typescript
export type CreateHook = (
  info: { functionName: FMember | FPMember; args: ArrayLike<unknown> }
) => boolean;
export type EndHook = (
  span: api.Span,
  info: { functionName: FMember | FPMember; args: ArrayLike<unknown>; error?: Error }
) => void;
```

OR split the info object up to separate arguments altogether. I'd support the latter I think if we're going to move span to be the first argument.